adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/preparedness/adapters/findcurrentlesson/CurrentLessonDatabaseAdapter.kt

```diff
@@ -0,0 +1,36 @@
+package com.krzykrucz.elesson.currentlesson.preparedness.adapters.findcurrentlesson
+
+import arrow.core.toOption
+import arrow.effects.IO
+import com.krzykrucz.elesson.currentlesson.attendance.domain.CheckedAttendanceList
+import com.krzykrucz.elesson.currentlesson.monolith.Database
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.FindCurrentLesson
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.UnpreparednessError
+import com.krzykrucz.elesson.currentlesson.shared.CurrentLesson
+import com.krzykrucz.elesson.currentlesson.shared.LessonAfterAttendance
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class AdapterConfig {
+    @Bean
+    fun getLessonBean(): FindCurrentLesson = {
+        Database.LESSON_DATABASE[it].toOption()
+            .toEither { UnpreparednessError.LessonNotStarted }
+            .map { persistentCurrentLesson ->
+
+                (if (persistentCurrentLesson.lessonTopic == null &&
+                    persistentCurrentLesson.attendance is CheckedAttendanceList) {
```

Comment:
this condition could be extracted to separated function to increased readability. Moreover, both checks could be implemented in  `PersistentCurrentLesson` itself as it has all data to answer those questions